### PR TITLE
energy: Add Monsoon Power Monitor Support

### PR DESCRIPTION
**Pull request by bjackman**
cc: @joelagnel

### CODE REVIEW

**Comment by derkling** [comment]
can we split this patch in a refactoring one, just for the AEP with a minimal changelog to explain what's going on, and a second patch which adds the monsoon support?

File: libs/utils/energy.py [hunk 1]
```diff
@@ -280,6 +263,25 @@ def report(self, out_dir, out_energy='energy.json', out_samples='samples.csv'):
...
 
+class AEP(_DevlibContinuousEnergyMeter):
+
+    def __init__(self, target, conf, res_dir):
+        super(AEP, self).__init__(target, res_dir)
+
+        # Configure channels for energy measurements
+        self._log.info('AEP configuration')
+        self._log.info('    %s', conf)
+        self._aep = devlib.EnergyProbeInstrument(
```
Comment (derkling) [inline libs/utils/energy.py]: We should now `s/self._aep/self._instrument/`

File: libs/utils/energy.py [hunk 2]
```diff
@@ -280,6 +263,25 @@ def report(self, out_dir, out_energy='energy.json', out_samples='samples.csv'):
...
+            self._target, labels=conf.get('channel_map'), **conf['conf'])
+
+        # Configure channels for energy measurements
+        self._log.debug('Enabling channels')
+        self._aep.reset()
```
Comment (derkling) [inline libs/utils/energy.py]: Same here

File: libs/utils/energy.py [hunk 3]
```diff
@@ -280,6 +263,25 @@ def report(self, out_dir, out_energy='energy.json', out_samples='samples.csv'):
...
+
+        # Logging enabled channels
+        self._log.info('Channels selected for energy sampling:')
+        self._log.info('   %s', str(self._aep.active_channels))
```
Comment (derkling) [inline libs/utils/energy.py]: And here

File: libs/utils/energy.py (hunk 1 above)
Comment (bjackman) [inline libs/utils/energy.py]: Oops, thanks!

**Comment by bjackman** [comment]
Fixed typo and pushed

**Comment by derkling** [comment]
Can you add on top also the notebook I've posted here:
https://github.com/derkling/lisa/commit/c845055a6588b0a753dce9c3db01cb7f885aa7ed

**Pull request merged**